### PR TITLE
Update the documentation after the BIO refactoring.

### PR DESCRIPTION
**Pull request by ciz**
The BIO was refactored in 417be660e1cd21a2ee085569ff98b0c4249b5416,
but the manual wasn't fully updated to reflect some of the changes.

##### Checklist
- [x] documentation is added or updated

### CODE REVIEW

File: doc/man3/BIO_s_connect.pod [hunk 1]
```diff
@@ -107,10 +109,10 @@ ports. This can be avoided by checking for the presence of the ':'
 character in the passed hostname and either indicating an error or
 truncating the string at that point.
 
-The values returned by BIO_get_conn_hostname(), BIO_get_conn_port(),
-BIO_get_conn_ip() and BIO_get_conn_int_port() are updated when a
-connection attempt is made. Before any connection attempt the values
-returned are those set by the application itself.
+The values returned by BIO_get_conn_hostname(), BIO_get_conn_address()
```
Comment (richsalz) [inline doc/man3/BIO_s_connect.pod]: comma at the end of line 112 (the Oxford comma:)

File: doc/man3/BIO_s_connect.pod [hunk 2]
```diff
@@ -124,9 +126,9 @@ then this is an indication that a connection attempt would block,
 the application should then take appropriate action to wait until
 the underlying socket has connected and retry the call.
 
-BIO_set_conn_hostname(), BIO_set_conn_port(), BIO_set_conn_ip(),
-BIO_set_conn_int_port(), BIO_get_conn_hostname(), BIO_get_conn_port(),
-BIO_get_conn_ip(), BIO_get_conn_int_port(), BIO_set_nbio() and
+BIO_set_conn_hostname(), BIO_set_conn_port(), BIO_get_conn_hostname(),
+BIO_set_conn_address(), BIO_set_conn_ip_family(), BIO_get_conn_port(),
+BIO_get_conn_address(), BIO_get_conn_ip_family(), BIO_set_nbio() and
```
Comment (richsalz) [inline doc/man3/BIO_s_connect.pod]: comma after ```BIO_set_nbio```

File: doc/man3/BIO_s_connect.pod [hunk 3]
```diff
@@ -136,21 +138,22 @@ BIO_s_connect() returns the connect BIO method.
 BIO_get_fd() returns the socket or -1 if the BIO has not
 been initialized.
 
-BIO_set_conn_hostname(), BIO_set_conn_port(), BIO_set_conn_ip() and
-BIO_set_conn_int_port() always return 1.
+BIO_set_conn_address(), BIO_set_conn_port() and BIO_set_conn_ip_family()
```
Comment (richsalz) [inline doc/man3/BIO_s_connect.pod]: comma after ```BIO_set_conn_port```

**Comment by ciz** [comment]
I've added the commas and also fixed the build travis failure (functions missing in the NAME section).

**Comment by richsalz** [comment]
Thanks for the fixes!

File: doc/man3/BIO_s_connect.pod [hunk 4]
```diff
@@ -4,6 +4,7 @@
 
 BIO_set_conn_address, BIO_get_conn_address,
 BIO_s_connect, BIO_new_connect, BIO_set_conn_hostname, BIO_set_conn_port,
+BIO_set_conn_ip_family, BIO_get_conn_ip_family
```
Comment (richsalz) [inline doc/man3/BIO_s_connect.pod]: trailing comma missing.

Comment (ciz) [reply doc/man3/BIO_s_connect.pod]: Haha. Damn you Oxford commas!
(one learns something new every day :)

Also, the *con_ip_family() functions aren't public according to the doc check, so I just remove them from the documentation.

Comment (richsalz) [reply doc/man3/BIO_s_connect.pod]: :)  Maybe @levitte and/or @kroeckx have a view on whether those should be public or not?

Comment (ciz) [reply doc/man3/BIO_s_connect.pod]: Unless @levitte or @kroeckx step in, this pull request should be ready to merge.

Comment (levitte) [reply doc/man3/BIO_s_connect.pod]: Errr, say what?  They are macro definitions present in `include/openssl/bio.h`, so the doc checker is faulty...  or rather, they need to be added in `util/private.num`.  Would you be willing to add them there and add them back in the docs?

Comment (ciz) [reply doc/man3/BIO_s_connect.pod]: Sure, I'll do that.

Comment (ciz) [reply doc/man3/BIO_s_connect.pod]: I added the BIO_{set,get}_conn_ip_family macros to both the docs and util/private.num.
I did the same for other "missing" macros, such as BIO_get_peer_name.

**Comment by mattcaswell** [comment]
Ping @richsalz.

**Comment by levitte** [comment]
Is anyone going to merge this?

**Comment by kaduk** [comment]
I'll merge

**Comment by kaduk** [comment]
Pushed to master (finally) -- thanks again for the fixes, and sorry for the delay.

**Comment by levitte** [comment]
No prob... I keep forgetting my gazillion branches as well, so you have my full understanding.

Thanks 🙂

**Pull request closed without merging**